src/Compilers/CSharp/Portable/Syntax/SyntaxKindFacts.cs

```diff
@@ -1048,9 +1050,15 @@ public static SyntaxKind GetPreprocessorKeywordKind(string text)
 
         public static IEnumerable<SyntaxKind> GetContextualKeywordKinds()
         {
-            for (int i = (int)SyntaxKind.YieldKeyword; i <= (int)SyntaxKind.WhenKeyword; i++)
+            ushort firstKeywordIdInclusive = (ushort)SyntaxKind.YieldKeyword;
+            ushort lastKeywordIdExclusive = (ushort)SyntaxKind.ElifKeyword;
+
+            for (ushort id = firstKeywordIdInclusive; id < lastKeywordIdExclusive; ++id)
             {
-                yield return (SyntaxKind)i;
+                if (Enum.IsDefined(typeof(SyntaxKind), id))
```

Comment:
IMO, because the old function was working quite well in a nice simple fashion, and the issue you ran into is suitably covered by a comment (which you need to add anyways).  Since you already need to add a comment, i'd prefer that that's all that we do.  So we're not fixing windows that aren't broken :)